src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -128,33 +133,29 @@ public Result getFirstResult() {
         return null;
     }
 
-    // TODO NR, SE: remove redundant setter after relationship change on client. Currently we need two deserializing setters for "result" (client) and "results" (server)
-    @JsonProperty(value = "result", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResult(Result result) {
-        this.setResults(result);
-    }
-
-    // TODO Ruscher, Entholzer: refactor to addResult
     /**
-     * custom setter that supports the migration from 1...1 to 1...* in the submission->result(s) relationship
-     * Will be refactore in the future
-     * @param result the result that should be added, in case this is null, an empty list will be used instead
+     * Used as a setResult method, as typically the latest result is used
+     *
+     * @param result
      */
-    @JsonProperty(value = "results", access = JsonProperty.Access.WRITE_ONLY)
-    public void setResults(Result result) {
-        if (result == null) {
-            // clear the list of results
+    // @JsonProperty(value = "result", access = JsonProperty.Access.WRITE_ONLY)
+    public void replaceLatestOrIfEmptyAddResult(Result result) {
```

Comment:
I think something is odd in this method. I am not seeing where you are adding the result for the case when `this.results == null`

Reply:
Yes, that's true, thanks for noticing. As it turns out the case `this.results == null` is never occurs. I'll remove it.

Reply:
In case of a newly created submission we have an empty ArrayList, in which case we have to add a result, or am I missing something?